src/lib.rs

```diff
@@ -1096,6 +1109,40 @@ impl GooseAttack {
             all_threads_logger = None;
         }
 
+        let throttle_receiver;
```

Comment:
would it be possible for this new code to be in a different method that returns `all_threads_throttle` , `parent_to_throttle_tx` ? This one method here is getting large

Reply:
Great suggestion, I've introduced `setup_throttle()` which simplified the logic as well.  (And in #100 I did a similar cleanup for `setup_logger()`).